testinfra/common/test_tor_mirror.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+
+def test_tor_mirror_present(File):
+    """
+    Ensure the FPF mirror of the Tor apt repo, tor-apt.ops.freedom.press,
+    is configured. This repository required manual updating with current
+    tor releases, to avoid breakage of untested updates.
+    """
+    f = File('/etc/apt/sources.list.d/tor_apt_ops_apt_freedom_press.list')
+
+    regex = ('^deb \[arch=amd64\] http:\/\/tor-apt\.ops\.freedom\.press '
+             'trusty main$')
+    assert f.contains(regex)
+
+
+def test_tor_mirror_fingerprint(Command):
+    """
+    Ensure the FPF tor mirror repo has the correct fingerprint on the
+    associated signing pubkey. We don't use the SecureDrop Release Signing
+    Key for the Tor mirror: packages are identical to the official Tor repo,
+    so signatures match, as well.
+    """
+    c = Command('apt-key finger')
+    tor_gpg_pub_key_info = """/etc/apt/trusted.gpg.d/deb.torproject.org-keyring.gpg
+-----------------------------------------------------
+pub   2048R/886DDD89 2009-09-04 [expires: 2020-08-29]
+      Key fingerprint = A3C4 F0F9 79CA A22C DBA8  F512 EE8C BC9E 886D DD89
+uid                  deb.torproject.org archive signing key
+sub   2048R/219EC810 2009-09-04 [expires: 2018-08-30]"""
+
+    assert c.rc == 0
+    assert tor_gpg_pub_key_info in c.stdout
+
+
+@pytest.mark.parametrize('filename', [
```

Comment:
this is probably a leftover of a previous implementation ?

Reply:
@dachary Can you clarify what "this" refers to? The test beginning on L36 of `testinfra/common/test_tor_mirror.py` checks for _absence_ of the official Tor apt repository, thereby complementing the tests that check for _presence_ of the FPF Tor mirror.

Reply:
the ``filename`` parameter is not used by the test below and it will fail testinfra